contrib/scripts/aws-iam-create-yubikey-mfa.sh

```diff
@@ -34,8 +34,17 @@ rm "$OUTFILE"
 
 CODE1=$(ykman oath code -s "$SERIAL_NUMBER")
 
-echo "Waiting 30s before generating a second code" >&2
-sleep 30
+# wait until next code can be generated
+# if SECONDS are :00 or :30, generate right away
+SECONDS=$(date +%S)
+if (( ${SECONDS#0} >= 1 && ${SECONDS#0} <= 29 )); then
+    WAIT_SECONDS=$(( 30 - ${SECONDS#0} ))
+elif (( ${SECONDS#0} >= 31 && ${SECONDS#0} <= 59 )); then
+    WAIT_SECONDS=$(( 60 - ${SECONDS#0} ))
+fi
+
+echo "Waiting ${WAIT_SECONDS:-0}s before generating a second code" >&2
+sleep ${WAIT_SECONDS:-0}
```

Comment:
can you create a function to encapsulate the logic to calculate the wait time

Reply:
done